### PR TITLE
Switch to KindToResource for ThirdPartyResourceData

### PR DESCRIPTION
**Pull request by brendandburns**
Ref https://github.com/kubernetes/kubernetes/issues/17435

@liggitt @lavalamp

### CODE REVIEW

File: pkg/registry/thirdpartyresourcedata/codec.go [hunk 1]
```diff
@@ -42,7 +43,9 @@ type thirdPartyResourceDataMapper struct {
 var _ meta.RESTMapper = &thirdPartyResourceDataMapper{}
 
 func (t *thirdPartyResourceDataMapper) isThirdPartyResource(resource string) bool {
-	return resource == strings.ToLower(t.kind)+"s"
+	plural, _ := meta.KindToResource(t.kind, false)
+	log.Printf(plural)
```
Comment (liggitt) [inline pkg/registry/thirdpartyresourcedata/codec.go]: Remove debug line

**Comment by lavalamp** [comment]
LGTM with fix to @liggitt's comment

**Comment by brendandburns** [comment]
@liggitt @lavalamp comment addressed, ptal.

Thanks!
--brendan

**Comment by liggitt** [comment]
LGTM

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Pull request merged**